### PR TITLE
Implement weather home screen UI

### PR DESCRIPTION
**Pull request by VitalinaSpinu**
Implement Weather Home Screen UI

### CODE REVIEW

File: Weather/Assets.xcassets/HomeScreen/Background.imageset/Contents.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "images" : [
     {
-      "filename" : "Image.svg",
+      "filename" : "photo-1436891620584-47fd0e565afb 1.png",
```
Comment (DmitriiVrabie) [inline Weather/Assets.xcassets/HomeScreen/Background.imageset/Contents.json]: use meaningful name

File: Weather/ContentView.swift [hunk 2]
```diff
@@ -6,15 +6,102 @@
...
         }
+        .background {
+            Image("Background")
+                .resizable()
+                .scaledToFill()
+                .ignoresSafeArea()
+        }
+    }
+}
+class WeatherViewModel: ObservableObject {
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: Extract to separate file

File: Weather/ContentView.swift [hunk 3]
```diff
@@ -6,15 +6,102 @@
...
+    
+    @Published var tempMax: String = "-"
+    @Published var tempMin: String = "-"
+    @Published var temp: String = "-"
+    @Published var description: String = "-"
+    
+    init() {
+        fetchWeather()
+    }
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: formatting

File: Weather/ContentView.swift [hunk 4]
```diff
@@ -6,15 +6,102 @@
...
+        let units = "metric"
+        let parameters = ["lat": lat, "lon": lon, "units": units, "appid": apiKey]
+        
+        AF.request(url, method: .get, parameters: parameters, encoding: URLEncoding.default, headers: nil, interceptor: nil)
+            .response { resp in
+                switch resp.result {
+                case .success(let data):
+                    do {
+                        let jsonData = try JSONDecoder().decode(WeatherDto.self, from: data!)
+                        print(jsonData)
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: Print shouldn't go to the main branch, please remove

File: Weather/ContentView.swift [hunk 5]
```diff
@@ -6,15 +6,102 @@
...
+                    } catch {
+                        print(error)
+                    }
+                case .failure(let error):
+                    print(error)
+                }
+            }
     }
 }
+struct WeatherDto: Codable {
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: extract to separate file

File: Weather/ContentView.swift [hunk 6]
```diff
@@ -6,15 +6,102 @@
...
+    let main: Main
+    let weather: [Weather]
+}
+
+struct Main: Codable {
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: extract to separate file

File: Weather/ContentView.swift [hunk 7]
```diff
@@ -6,15 +6,102 @@
...
+    
+    init() {
+        fetchWeather()
+    }
+    func fetchWeather() {
+        let url = "https://api.openweathermap.org/data/2.5/weather"
+        let apiKey:String = "4f071ac6d017b2bdfc07918768d91cb3"
+        let lat = "47.003670"
+        let lon = "28.907089"
+        let units = "metric"
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: Please use constants for fields that won't be changed

File: Weather/ContentView.swift [hunk 8]
```diff
@@ -6,15 +6,102 @@
...
+        let parameters = ["lat": lat, "lon": lon, "units": units, "appid": apiKey]
+        
+        AF.request(url, method: .get, parameters: parameters, encoding: URLEncoding.default, headers: nil, interceptor: nil)
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: Please extract this to separate class to reuse it for other requests

File: Weather/ContentView.swift [hunk 9]
```diff
@@ -6,15 +6,102 @@
...
+            Image("Background")
+                .resizable()
+                .scaledToFill()
+                .ignoresSafeArea()
+        }
+    }
+}
+class WeatherViewModel: ObservableObject {
+    
+    @Published var tempMax: String = "-"
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: Why not blank string but "-"?

File: Weather/ContentView.swift [hunk 10]
```diff
@@ -6,15 +6,102 @@
...
+    func fetchWeather() {
+        let url = "https://api.openweathermap.org/data/2.5/weather"
+        let apiKey:String = "4f071ac6d017b2bdfc07918768d91cb3"
+        let lat = "47.003670"
+        let lon = "28.907089"
+        let units = "metric"
+        let parameters = ["lat": lat, "lon": lon, "units": units, "appid": apiKey]
+        
+        AF.request(url, method: .get, parameters: parameters, encoding: URLEncoding.default, headers: nil, interceptor: nil)
+            .response { resp in
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: We need to show to the user loading while we are fetching weather data

File: Weather/ContentView.swift [hunk 11]
```diff
@@ -6,15 +6,102 @@
...
+                    .foregroundStyle(.white)
+                    .frame(height: 70)
+                Text(viewModel.description)
+                    .font(.system(size: 20) .weight(.bold))
+                    .foregroundStyle(.gray)
+            }
+            HStack {
+                Spacer()
+                Text("H: \(viewModel.tempMax)°")
+                    .font(.system(size: 20) .weight(.bold))
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift]: Font size could be extracted to constant to reuse

**Review by DmitriiVrabie — CHANGES_REQUESTED** [review]
Fix commentaries

File: Weather/ContentView.swift (hunk 7 above)
Comment (VitalinaSpinu) [reply Weather/ContentView.swift]: Here everything is constant

File: Weather/Assets.xcassets/HomeScreen/Background.imageset/Contents.json (hunk 1 above)
Comment (VitalinaSpinu) [reply Weather/Assets.xcassets/HomeScreen/Background.imageset/Contents.json]: I did not create this file and I did not give this name. I put the images in the Assets file and then give them meaningful names. Please check!

File: Weather/ContentView.swift (hunk 3 above)
Comment (VitalinaSpinu) [reply Weather/ContentView.swift]: I didn't understand what to do. Please explain to me.

File: Weather/ContentView.swift (hunk 10 above)
Comment (VitalinaSpinu) [reply Weather/ContentView.swift]: I'm sorry, but I didn't understand what to do here either. Please explain to me.

File: Weather/ContentView.swift [hunk 12]
```diff
@@ -8,13 +8,50 @@
 import SwiftUI
 
 struct ContentView: View {
+    
+    @StateObject var viewModel = WeatherViewModel()
+    let fontSize = 20
+    var yOffset: CGFloat = 50
```
Comment (DmitriiVrabie) [inline Weather/ContentView.swift:14]: Please remove if not used

**Review by DmitriiVrabie — APPROVED** [review]

**Pull request merged**